### PR TITLE
Modify rule S4619: Adapt to LaYC

### PR DESCRIPTION
**Pull request by victor-diez-sonarsource**

### CODE REVIEW

File: rules/S4619/javascript/rule.adoc [hunk 1]
```diff
@@ -1,42 +1,26 @@
...
-    if (expectedValue in arr) { // Noncompliant, will be always true because the array is made of 3 elements and the #1 is always there whatever its value
+    if (expectedValue in arr) { // Noncompliant: will be always false
         return expectedValue + " found in the array";
     } else {
         return expectedValue + " not found";
     }
 }
 ----
 
+Use the Array method `includes()` to determine whether an array contains a certain value. If the actual intention is to check for an array slot, use ``++hasOwnProperty++``.
```
Comment (yassin-kammoun-sonarsource) [inline rules/S4619/javascript/rule.adoc:21]:
```suggestion
Use the method `Array.prototype.includes()` to determine whether an array contains a certain value. If the actual intention is to check for an array slot, use ``++ Object.prototype.hasOwnProperty() ++``.
```

Comment (yassin-kammoun-sonarsource) [reply rules/S4619/javascript/rule.adoc:21]: Not convinced about this suggestion. Feel free to ignore it.

File: rules/S4619/javascript/rule.adoc [hunk 2]
```diff
@@ -1,42 +1,26 @@
...
 
-=== Compliant solution
-
-[source,javascript]
+[source,javascript,diff-id=1,diff-type=compliant]
 ----
 function func() {
     let arr = ["a", "b", "c"];
```
Comment (yassin-kammoun-sonarsource) [inline rules/S4619/javascript/rule.adoc:26]: Let's maybe use `const` instead.

File: rules/S4619/javascript/rule.adoc [hunk 3]
```diff
@@ -1,42 +1,26 @@
...
-If checking for an array slot is indeed desired, using ``++hasOwnProperty++`` makes the code intention clearer.
-
-
-=== Noncompliant code example
-
-[source,javascript]
+[source,javascript,diff-id=1,diff-type=noncompliant]
 ----
 function func1() {
     let arr = ["a", "b", "c"];
```
Comment (yassin-kammoun-sonarsource) [inline rules/S4619/javascript/rule.adoc:10]: Let's maybe use `const` instead.

**Review by yassin-kammoun-sonarsource — APPROVED** [review]
LGTM.

File: rules/S4619/javascript/rule.adoc (hunk 1 above)
Comment (victor-diez-sonarsource) [reply rules/S4619/javascript/rule.adoc:21]: what would you suggest?

Comment (yassin-kammoun-sonarsource) [reply rules/S4619/javascript/rule.adoc:21]: I meant I am not convinced by what I suggested. What you wrote is fine :)

Comment (victor-diez-sonarsource) [reply rules/S4619/javascript/rule.adoc:21]: ah, ok. I like it, I will change it

**Pull request closed without merging**